bot/cogs/moderation/silence.py

```diff
@@ -118,12 +151,17 @@ async def _silence(self, channel: TextChannel, persistent: bool, duration: Optio
         `duration` is only used for logging; if None is passed `persistent` should be True to not log None.
         Return `True` if channel permissions were changed, `False` otherwise.
```

Comment:
> `duration` is only used for logging

I feel like the log could be done on the outside, so that the param is unnecessary (we return the bool to signal success).

Additionally, I feel like the information represented by `persistent` and `duration` can be expressed by one value, since `duration` is already optional; if it's `None` we know that the silence is indefinite, right? I also find `persistent` to be an awkward name - on the surface it gives me the impression of telling whether the silence should be persisted.

I understand that this is not part of this PR's diff, but figured I'd mention it anyway.

Reply:
My comment here is mainly motivated by the fact that one param restricts the valid values of the other, i.e.:
> if None is passed `persistent` should be True to not log None

I generally try to avoid that and see if I can find a way to represent the information using a single value that cannot exist in an invalid state. Here I think the solution is fairly easy, unless I'm missing something.

Reply:
I refactored it in cf2c032

Reply:
Yes, nice. I think that's much better.